library/src/scala/tasty/reflect/Kernel.scala

```diff
@@ -1067,6 +1067,12 @@ trait Kernel {
   /** Source code within the position */
   def Position_sourceCode(self: Position): String
 
+  /** Create a position that ranges from start to end (exclusive) in the given source
+   *  @param start index of the start of the range (0 <= start < end)
+   *  @param end index of the end of the range (start < end <= sizeOf(pos.sourceFile))
+   */
+  def Position_withOffset(self: Position)(start: Int, end: Int): Position
```

Comment:
Then I see this API, the first guess is that the offset is _relative_ to the given position `self`. But in the implementation, only the source file is used. It seems this is better to be defined as a constructor for `Position` which takes sourceFile directly.

Reply:
That would be terrible, each source position will try to reload the file.

Reply:
Maybe we should have the source file abstraction in TASTy reflect